pages/student/testing-page.php

```diff
@@ -0,0 +1,232 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Testing Page</title>
+    <link rel="stylesheet" href="../../style.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css">
+</head> 
+<body>
+    <header>
+        <nav>
+            <ul>
+                <li><a href="#">Home</a></li>
+                <li><a href="#">Stats</a></li>
+                <li><a href="#">Leaderboards</a></li>
+            </ul>
+        </nav>
+    </header>
+    <?php 
+    require_once('../../php/connection.php');
+    session_start();
+
+    $userID = $_SESSION["userID"];
+    $conn = newConn();
+
+    //Get the logged in user's details, things like the first name and studentID
+    $stmt = $conn->prepare("CALL GetStudentData(:userID)");
+    $stmt->bindValue(":userID", $userID, PDO::PARAM_STR);
+    $stmt->execute();
+    $studentData = $stmt->fetch(PDO::FETCH_ASSOC);
+
+    $userName = $studentData['firstname'];
+    $studentID = $studentData['studentID'];
+
+   //Query to get the tests
+   $stmt = $conn->prepare("CALL GetStudentsQuizzes(:studentID)");
+   $stmt->bindValue(":studentID", $studentID, PDO::PARAM_STR);
+   $stmt->execute();
+   $allTests = $stmt->fetchAll();
+
+   if ($_SERVER['REQUEST_METHOD'] == 'POST') {
+    $quizID = $_POST['quizID'];
+    $studentQuizLinkID = $_POST['studentQuizLinkID'];
+
+    //echo "Student Quiz Link ID: " . $studentQuizLinkID;
+
+    //Get Quiz Data
+    $stmt = $conn->prepare("CALL GetQuizData(:quizID)");
+    $stmt->bindParam(":quizID", $quizID, PDO::PARAM_STR);
+    $stmt->execute();
+    $quizData = $stmt->fetch(PDO::FETCH_ASSOC);
+
+    $subjectName = $quizData['subject'];
+    $quizName = $quizData['title'];
+
+    $subjectName = $quizData['subject'];
+    $quizName = $quizData['title'];
+
+    //Get Quiz Set
+    $stmt = $conn->prepare("CALL GetAllQuestions(:quizID)");
+    $stmt->bindParam(":quizID", $quizID, PDO::PARAM_STR);
+    $stmt->execute();
+    $allQuestions = $stmt->fetchAll();
+
+    //Get question number and total number of questions
+    $questionNumber = 0;
+    $totalQuestions = count($allQuestions);
+}
```

Comment:
I would heavily suggest using `GetQuizSet` to get a set a questions, as this randomises the order and allows you to limit how many questions are in each quiz. However, what you have now is fine.

Reply:
I originally did use GetQuizSet, but I was having an error with it the other day so I switched to GetAllQuestions. Have switched back and it now works